lib/features/composer/presentation/widgets/sending_message_dialog_view.dart

```diff
@@ -0,0 +1,240 @@
+import 'dart:async';
+import 'dart:math';
+
+import 'package:core/presentation/extensions/capitalize_extension.dart';
+import 'package:core/presentation/extensions/color_extension.dart';
+import 'package:core/presentation/state/failure.dart';
+import 'package:core/presentation/state/success.dart';
+import 'package:core/presentation/views/button/tmail_button_widget.dart';
+import 'package:core/utils/app_logger.dart';
+import 'package:dartz/dartz.dart' as dartz;
+import 'package:dio/dio.dart';
+import 'package:flutter/material.dart';
+import 'package:get/get.dart';
+import 'package:tmail_ui_user/features/composer/domain/exceptions/compose_email_exception.dart';
+import 'package:tmail_ui_user/features/composer/domain/state/generate_email_state.dart';
+import 'package:tmail_ui_user/features/composer/domain/state/send_email_state.dart';
+import 'package:tmail_ui_user/features/composer/domain/usecases/create_new_and_send_email_interactor.dart';
+import 'package:tmail_ui_user/features/composer/presentation/model/create_email_request.dart';
+import 'package:tmail_ui_user/main/exceptions/remote_exception.dart';
+import 'package:tmail_ui_user/main/localizations/app_localizations.dart';
+import 'package:tmail_ui_user/main/routes/route_navigation.dart';
+
+typedef OnCancelSendingEmailAction = Function({CancelToken? cancelToken});
+
+class SendingMessageDialogView extends StatefulWidget {
+
+  final CreateEmailRequest createEmailRequest;
+  final CreateNewAndSendEmailInteractor createNewAndSendEmailInteractor;
+  final OnCancelSendingEmailAction? onCancelSendingEmailAction;
+  final CancelToken? cancelToken;
+
+  const SendingMessageDialogView({
+    super.key,
+    required this.createEmailRequest,
+    required this.createNewAndSendEmailInteractor,
+    this.onCancelSendingEmailAction,
+    this.cancelToken,
+  });
+
+  @override
+  State<SendingMessageDialogView> createState() => _SendingMessageDialogViewState();
+}
+
+class _SendingMessageDialogViewState extends State<SendingMessageDialogView> {
+
+  StreamSubscription? _streamSubscription;
+  final ValueNotifier<dartz.Either<Failure, Success>?> _viewStateNotifier = ValueNotifier(null);
+
+  @override
+  void initState() {
+    super.initState();
+    _streamSubscription = widget.createNewAndSendEmailInteractor
+      .execute(
+        createEmailRequest: widget.createEmailRequest,
+        cancelToken: widget.cancelToken
+      )
+      .listen(
+        _handleDataStream,
+        onError: _handleErrorStream
+      );
+  }
+
+  void _handleDataStream(dartz.Either<Failure, Success> newState) {
+    _viewStateNotifier.value = newState;
+
+    newState.fold(
+      (failure) {
+        if (failure is SendEmailFailure || failure is GenerateEmailFailure) {
+          popBack(result: failure);
+        }
+      },
```

Comment:
idem

Reply:
We do not have other failures in this case.